spark-java/src/main/java/com/baeldung/sparkjava/UserServiceMapImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.sparkjava;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+public class UserServiceMapImpl  implements UserService{
+    private HashMap<String, User> userMap;
+    
+    public UserServiceMapImpl() {
+        userMap = new HashMap<>();
+    }
+
+    public void addUser (User user) {
+        userMap.put(user.getId(), user);
+    }
+    
+    public Collection<User> getUsers () {
+        return  userMap.values();
+    }
+    
+    public User getUser (String id) {
+        return userMap.get(id);
+    }
+    
+    public User editUser (String id, HashMap userArg) throws Exception{
+        User toEdit = userMap.get(id);
+        if (toEdit == null )
+            return null;
+        toEdit.setEmail((userArg.get("email")!=null) ? (String) userArg.get("email") : toEdit.getEmail() );
```

Comment:
This comment applies to the next three lines as well if you stick with a Map implementation:
Notice that when the key is not found in the map, you are unnecessarily setting the property to the value that it already has. Just use 'if' statements and set the properties only when the key exists in the map.